griptape/tools/google_docs/tool.py

```diff
@@ -0,0 +1,270 @@
+from __future__ import annotations
+from typing import TYPE_CHECKING
+import logging
+from attr import field, define
+from schema import Schema, Optional, Literal
+from griptape.artifacts import ErrorArtifact, InfoArtifact, BlobArtifact, ListArtifact
+from griptape.utils.decorators import activity
+from griptape.tools import BaseGoogleClient
+
+if TYPE_CHECKING:
+    from googleapiclient.discovery import Resource
+
+@define
+class GoogleDocsClient(BaseGoogleClient):
+    DOCS_SCOPES = ["https://www.googleapis.com/auth/documents"]
+
+    DRIVE_SCOPES = ["https://www.googleapis.com/auth/drive.file"]
+
+    owner_email: str = field(default=None)
+
+    @activity(
+        config={
+            "description": "Can be used to create a new Google Doc in Drive",
+            "schema": Schema(
+                {
+                    Literal(
+                        "file_name",
+                        description="Name of the file to be created."
+                    ): str
+                }
+            ),
+        }
+    )
+    def create_google_doc(self, params: dict) -> InfoArtifact | ErrorArtifact:
+        try:
+            file_name = params["file_name"]
+            service = self._build_client(self.DOCS_SCOPES, "v1", "docs")
+            doc = service.documents().create(body={"title": file_name}).execute()
+            return InfoArtifact(f"Google Doc '{file_name}' created with ID: {doc['documentId']}")
+
+        except Exception as e:
+            logging.error(e)
+            return ErrorArtifact(f"Error creating Google Doc: {e}")
+
+    @activity(
+        config={
+            "description": "Can be used to append text to a Google Doc",
+            "schema": Schema(
+                {
+                    Literal(
+                        "file_path",
+                        description="Destination file path on Google Drive in the POSIX format. "
+                                    "For example, 'foo/bar/baz.txt'"
+                    ): str,
+                    Literal(
+                        "text",
+                        description="Text to be appended to the Google Doc."
+                    ): str
+                }
+            ),
+        }
+    )
+    def append_text_to_google_doc(self, params: dict) -> InfoArtifact | ErrorArtifact:
+        try:
+            file_path = params["file_path"]
+            text = params["text"]
+            service = self._build_client(self.DOCS_SCOPES, "v1", "docs")
+            drive_service = self._build_client(self.DRIVE_SCOPES, "v3", "drive")
+            document_id = self._path_to_file_id(drive_service, file_path)
+            if document_id:
+                doc = service.documents().get(documentId=document_id).execute()
+                content = doc["body"]["content"]
+                last_text = content[-1]["paragraph"]["elements"][-1]["textRun"]["content"]
```

Comment:
What is content has no paragraphs? What if a paragraph has no elements?

Reply:
it still does append and prepend even if there's no text in the file